### PR TITLE
Cake Cat Cak - How the Chef turned Frankenstein with cake base, sp(r)inkles and frosting

### PR DESCRIPTION
**Pull request by Terilia**
This is a port with minor adjustments of [Xhuis](https://github.com/Xhuis) [Caks pet from TG Station](https://github.com/tgstation/tgstation/pull/18483)

- 50 max health
  -  Regenerates brute damage at a rate of 4/tick 
- Takes 10 brute damage when hit with a harm intent attack
  -  Harm intent attacks take a bite from the cak, adding nutriment and vitamin to the attacker
- Created by crafting a brain, heart, birthday cake, three slabs of meat, 30u of blood, 5u of sprinkles, and 1u of teslium
  -  If the brain is occupied by a player, the player is placed in control of the cak
- Butchering drops some (but not all) of the components used in creating the cak
  -  Brains dropped in this manner lose sentience

Updated the text as the mechanics have been changed back. 

Extra: During porting I noticed that frosted donuts contained "spinkles" instead of "sprinkles" - I understand that making 2 PR's is normally the way to go, but considering that this is a very very minor fix, I hope that for this one occasion, it will be fine. 

image

:cl:
add: Added Cake Cat - allowing the chef to create a living small pet in his kitchen. 
fix: Fixed frosted donuts, as they were filled with "spinkles" and not "sprinkles"
/:cl:

### CODE REVIEW

File: code/modules/food_and_drinks/food/foods/baked_goods.dm [hunk 1]
```diff
@@ -345,7 +345,7 @@
 /obj/item/reagent_containers/food/snacks/donut/sprinkles
 	name = "frosted donut"
 	icon_state = "donut2"
-	list_reagents = list("nutriment" = 3, "sugar" = 2, "spinkles" = 2)
+	list_reagents = list("nutriment" = 3, "sugar" = 2, "sprinkles" = 2)
```
Comment (Fox-McCloud) [inline code/modules/food_and_drinks/food/foods/baked_goods.dm]: good old spinkles.

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 2]
```diff
@@ -177,3 +177,92 @@
...
+	icon_living = "cak"
+	icon_resting = "cak_rest"
+	icon_dead = "cak_dead"
+	health = 50
+	maxHealth = 50
+	harm_intent_damage = 10
+	butcher_results = list(
+		/obj/item/organ/internal/heart = 1, 
+		/obj/item/reagent_containers/food/snacks/birthdaycakeslice = 3,  
+		/obj/item/reagent_containers/food/snacks/meat = 2
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: should drop meat as well.

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 3]
```diff
@@ -177,3 +177,92 @@
...
+
+		return 1
+
+
+	if(istype(O, /obj/item/kitchen/knife))
+		to_chat(user, "<span class='warning'>You cut out the brain from [src].</span>")
+		eject_brain()
+		return 1
+
+	else if(user.a_intent == INTENT_HARM && user.reagents)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: Not where this should be.

This is attackby, not attackhand.

File: code/modules/food_and_drinks/recipes/tablecraft/recipes_table.dm [hunk 4]
```diff
@@ -252,4 +252,17 @@
 	)
 	tools = list(/obj/item/kitchen/sushimat)
 	result = /obj/item/reagent_containers/food/snacks/sliceable/Tai_maki
+	category = CAT_FOOD
+
+/datum/crafting_recipe/food/cak
+	name = "Cake Cat"
```
Comment (Fox-McCloud) [inline code/modules/food_and_drinks/recipes/tablecraft/recipes_table.dm]: `name = "Living cat/cake hybrid"`

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 5]
```diff
@@ -177,3 +177,92 @@
...
+						break
+				for(var/mob/living/simple_animal/S in GLOB.player_list)
+					if(S in GLOB.respawnable_list)
+						ghost_can_reenter = 1
+						break
+			if(!ghost_can_reenter)
+				to_chat(user, "<span class='notice'>[B] is completely unresponsive; there's no point.</span>")
+				return
+
+		to_chat(user, "<span class='notice'>You install [B] in [src]!</span>")
```
Comment (taukausanake) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: I think install is a little weird for a food being. Can't just say "stick into" instead?

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 6]
```diff
@@ -177,3 +177,92 @@
...
+	butcher_results = list(
+		/obj/item/organ/internal/heart = 1, 
+		/obj/item/reagent_containers/food/snacks/birthdaycakeslice = 3,  
+		/obj/item/reagent_containers/food/snacks/meat = 2
+	)
+	response_harm = "takes a bite out of"
+	attacked_sound = "sound/items/eatfood.ogg"
+	deathmessage = "loses its false life and collapses!"
+	death_sound = "bodyfall"
+	var/obj/item/organ/internal/brain/brain = null
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: I really think utilizing the built in functionality of `CheckParts` would be better than this, to be honest.

```
/mob/living/simple_animal/pet/cat/cak/CheckParts(list/parts)
	..()
	var/obj/item/organ/brain/B = locate(/obj/item/organ/brain) in contents
	if(!B || !B.brainmob || !B.brainmob.mind)
		return
	B.brainmob.mind.transfer_to(src)
	to_chat(src, "<span class='big bold'>You are a cak!</span><b> You're a harmless cat/cake hybrid that everyone loves. People can take bites out of you if they're hungry, but you regenerate health \
	so quickly that it generally doesn't matter. You're remarkably resilient to any damage besides this and it's hard for you to really die at all. You should go around and bring happiness and \
	free cake to the station!</b>")
	var/new_name = stripped_input(src, "Enter your name, or press \"Cancel\" to stick with Keeki.", "Name Change")
	if(new_name)
		to_chat(src, "<span class='notice'>Your name is now <b>\"new_name\"</b>!</span>")
		name = new_name
```

**Comment by taukausanake** [comment]
Can you just make cake cat NPC simple animals that are not player controlled? I don't know if I'm reading the code correctly but I don't think there's anything that says the brain doesn't have to have a mind in it

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 7]
```diff
@@ -177,3 +177,56 @@
...
+	icon_resting = "cak_rest"
+	icon_dead = "cak_dead"
+	health = 50
+	maxHealth = 50
+	harm_intent_damage = 10
+	butcher_results = list(
+		/obj/item/organ/internal/heart = 1, 
+		/obj/item/reagent_containers/food/snacks/birthdaycakeslice = 3,  
+		/obj/item/reagent_containers/food/snacks/meat/slab = 2
+	)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: don't forget to make it drop a brain.

**Comment by datlo** [comment]
I'm still confused by this idea but I see nothing wrong with adding it if the code issues are fixed

File: code/modules/food_and_drinks/recipes/tablecraft/recipes_table.dm [hunk 8]
```diff
@@ -252,4 +252,18 @@
 	)
 	tools = list(/obj/item/kitchen/sushimat)
 	result = /obj/item/reagent_containers/food/snacks/sliceable/Tai_maki
+	category = CAT_FOOD
+
+/datum/crafting_recipe/food/cak
```
Comment (Fox-McCloud) [inline code/modules/food_and_drinks/recipes/tablecraft/recipes_table.dm]: To get the brain transferring to work, you need to add this, above the cak's crafting recipe:

```
/datum/crafting_recipe/food

/datum/crafting_recipe/food/New()
	parts |= reqs
```

File: code/modules/food_and_drinks/recipes/tablecraft/recipes_table.dm [hunk 9]
```diff
@@ -252,4 +252,18 @@
...
+		/obj/item/organ/internal/brain = 1,
+		/obj/item/organ/internal/heart = 1,
+		/obj/item/reagent_containers/food/snacks/sliceable/birthdaycake  = 1,
+		/obj/item/reagent_containers/food/snacks/meat/slab  = 3,
+		/datum/reagent/blood = 30,
+		/datum/reagent/consumable/sprinkles = 5,
+		/datum/reagent/teslium = 1,
+	)
+	result = /mob/living/simple_animal/pet/cat/cak
 	category = CAT_FOOD
```
Comment (Fox-McCloud) [inline code/modules/food_and_drinks/recipes/tablecraft/recipes_table.dm]: You can get rid of this, since it'll inherit this category for the food parent.

**Comment by Aurorablade** [comment]
I demand a feral varient based on the feral cat mob.

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 10]
```diff
@@ -177,3 +177,57 @@
...
+	for(var/obj/item/reagent_containers/food/snacks/donut/D in range(1, src))
+		if(D.icon_state != "donut2")
+			D.name = "frosted donut"
+			D.icon_state = "donut2"
+			D.reagents.add_reagent("sprinkles", 2)
+			D.filling_color = "#FF69B4"
+
+/mob/living/simple_animal/pet/cat/cak/attack_hand(mob/living/L)
+	..()
+	if(L.a_intent == "harm" && L.reagents && !stat)
```
Comment (Ty-Omaha) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: Using the define would be better

`L.a_intent == INTENT_HARM`

**Pull request merged**